### ISSUE TITLE
Getting very high MAP values

### ISSUE THREAD

**Issue by shuklam20** [Cartucho/mAP#23]
Hi,
I have my own data on which I have trained YOLO and trying to use your method to verify the MAP values, however, I am obtaining a result of ~98% which is too high (I calculated it using darknet v3 repo:  [https://github.com/AlexeyAB/darknet#how-to-calculate-map-on-pascalvoc-2007](link) as well, which gave me a result of ~80%). I can see that there are FP and TN from the visualizations so this high value seems weird. From top of your head, can you think of what might be the reason? I am attaching snaps of some results.
![ground-truth info](https://user-images.githubusercontent.com/18172899/42107045-04c426b6-7ba4-11e8-94c7-b41e2d0670ae.png)
![map](https://user-images.githubusercontent.com/18172899/42107046-04e2c67a-7ba4-11e8-8b3c-bfd63853f1b0.png)
![predicted objects info](https://user-images.githubusercontent.com/18172899/42107047-04f0c482-7ba4-11e8-840d-3ccf780637df.png)

**Comment by Cartucho** [comment]
Can you add here the snaps of the PR curves as well? That's what counts the most

**Comment by Cartucho** [comment]
Btw, I just noticed that it seems that you trained the class `tool` for example and it is not present in the ground-truth.
Could you also put here a printscreen of the mAP calculated with the darknet repo?

**Comment by shuklam20** [comment]
Hey, sorry for I didn't get back to you earlier. I had run the model in my office computer and had no access to it over the weekend. Here are P-R Curves for some of the classes:
![can](https://user-images.githubusercontent.com/18172899/42167868-559641ce-7ddd-11e8-9ff6-dba0045d07ad.png)
![hangtag](https://user-images.githubusercontent.com/18172899/42167869-55ab4844-7ddd-11e8-9dc7-ac8e73a9a8c7.png)
![bag](https://user-images.githubusercontent.com/18172899/42167923-7b2aa0d8-7ddd-11e8-95f9-2a5eea295c1d.png)

I rechecked with another code that I wrote and got a similar MAP. So I dug through my code to discover that I had collected the predictions at a threshold of 0.05 which is super low and hence it literally outputted a lot of classes, giving such high values for the MAP. However, thinking about the MAP's formula, shouldn't low threshold increase false positives as well? Does that not impact a lot on its value? If low threshold gives a good MAP value, then is MAP really a good measure for object detection! Am I missing something? My apologies if this is not the correct platform to ask these questions.
